src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -392,6 +406,26 @@ def decode(self, embed_ind):
         quantize = quantize.permute(0, 2, 1)
         return quantize
 
+    def forward(self, hidden_states):
+        device = hidden_states.device
+
+        hidden_states = hidden_states.permute(0, 2, 1)
+
+        embed_ind, quantize = self.codebook(hidden_states)
+
+        loss = torch.tensor([0.0], device=device, requires_grad=self.training)
+
+        if self.training:
+            # Pass the gradients straight through the quantization, by directly linking the gradients of the input
+            # embed_ind with the output quantize in the computation graph.
+            quantize = embed_ind + (quantize - embed_ind).detach()
+
+            commitment_loss = nn.functional.mse_loss(quantize.detach(), embed_ind)
+            loss = loss + commitment_loss
+
+        quantize = quantize.permute(0, 2, 1)
+        return loss, embed_ind, quantize
```

Comment:
Can we return an instance of `ModelOutput`, for example like we do in MBart:
https://github.com/huggingface/transformers/blob/134caef31a40eff778bb8dc7b917dac9d7e3177a/src/transformers/models/mbart/modeling_mbart.py#L1385